test/CheckedCRewriter/allocator.c

```diff
@@ -0,0 +1,50 @@
+// Tests for Checked C rewriter tool.
+//
+// Tests for malloc and friends. 
+//
+// RUN: checked-c-convert %s -- | FileCheck -match-full-lines %s
+// RUN: checked-c-convert %s -- | %clang_cc1 -fno-builtin -verify -fcheckedc-extension -x c -
+// expected-no-diagnostics
+//
+typedef __SIZE_TYPE__ size_t;
+extern void *malloc(size_t);
+extern void free(void *);
+
+void dosomething(void) {
+  int a = 0;
+  int *b = &a;
+  *b = 1;
+  return;
+}
+
+void foo(void) {
+  int *a = (int *) malloc(sizeof(int));
+  *a = 0;
+  free(a);
+  return;
+}
+//CHECK: void foo(void) {
+//CHECK-NEXT: _Ptr<int>  a = (int *) malloc(sizeof(int));
+//CHECK-NEXT: *a = 0;
+//CHECK-NEXT: free((void *)a);
+
+typedef struct _listelt {
+  struct _listelt *next;
+  int val;
+} listelt;
+
+typedef struct _listhead {
+  listelt *hd;
+} listhead;
+
+void add_some_stuff(listhead *hd) {
```

Comment:
Well If my understanding of `checked` blocks is correct, `ptr<T> var = (T*)malloc(sizeof(T));` would be disallowed in a checked block, but `ptr<T> var = (ptr<T>)malloc(sizeof(T));` would be allowed, suggesting if we want programmers to do as few changes as possible, then we should do the latter.

(Eventually) It doesn't seem beyond the bounds of possibility to have the checked c convert tool add `checked` keywords to functions that only use checked pointers (ie, functions that only use `T*` like `ptr<T>`, as that's the only analysis you're currently doing), and so preempting this doesn't seem like a bad idea, if it's a quick change.